core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -970,11 +970,25 @@ private static String getCategoryDisplayName(String category) {
 
     public List<Plugin> getUpdates() {
         Map<String,Plugin> pluginMap = new LinkedHashMap<>();
+        Map<String, List<Plugin>> incompatiblePluginMap = new LinkedHashMap<>();
+
         for (UpdateSite site : sites) {
             for (Plugin plugin: site.getUpdates()) {
                 final Plugin existing = pluginMap.get(plugin.name);
                 if (existing == null) {
                     pluginMap.put(plugin.name, plugin);
+
+                    if (!plugin.isNeededDependenciesCompatibleWithInstalledVersion()) {
+                       for (Plugin incompatiblePlugin : plugin.getDependenciesIncompatibleWithInstalledVersion()) {
+                           if (!incompatiblePluginMap.containsKey(incompatiblePlugin.name)) {
```

Comment:
This can be simplified using `incompatiblePluginMap.computeIfAbsent`.

Reply:
Oh thanks, I didn't know that method!